modules/auxiliary/gather/external_ip.rb

```diff
@@ -29,12 +29,19 @@ def initialize
     register_options(
       [
         Opt::RHOST('ifconfig.me'),
+        OptString.new('VHOST', [false, "HTTP server hostname", 'ifconfig.me']),
         OptBool.new('REPORT_HOST', [false, 'Add the found IP to the database', false])
       ])
 end
 
   def run
     connect
+
+    # Use VHOST when set.
+    def rhost
+      datastore['VHOST'] || datastore['RHOST']
+    end
```

Comment:
`rhost` should be the `RHOST` not the `vhost`, it's used by quite a few mixins so this should not be changed.

Reply:
renaming `def rhost` to `def vhost` and dropping the `datastore['RHOST']`, will be sufficient?
because leaving the `datastore['RHOST']` alone won't make the difference due to https://github.com/rapid7/metasploit-framework/issues/9778

Reply:
Yeah, I tested it and the only thing you need to do is set the default value for the VHOST header. You can completely remove this change.

This was done from the master branch without these changes to show that `VHOST` is the only thing that needs to be set.
```
msf6 auxiliary(gather/external_ip) > show options 

Module options (auxiliary/gather/external_ip):

   Name         Current Setting  Required  Description
   ----         ---------------  --------  -----------
   Proxies                       no        A proxy chain of format type:host:port[,type:host:port][...]
   REPORT_HOST  false            no        Add the found IP to the database
   RHOSTS       ifconfig.me      yes       The target host(s), range CIDR identifier, or hosts file with syntax 'file:<path>'
   RPORT        80               yes       The target port (TCP)
   SSL          false            no        Negotiate SSL/TLS for outgoing connections
   VHOST                         no        HTTP server virtual host

msf6 auxiliary(gather/external_ip) > run
[*] Running module against 216.239.32.21
[*] Running module against 216.239.34.21
[*] Running module against 216.239.36.21
[*] Running module against 216.239.38.21
[*] Auxiliary module execution completed
msf6 auxiliary(gather/external_ip) > set VHOST ifconfig.me
VHOST => ifconfig.me
msf6 auxiliary(gather/external_ip) > run
[*] Running module against 216.239.38.21

[+] Source ip to 216.239.38.21 is ###.###.220.23
[*] Running module against 216.239.32.21
[+] Source ip to 216.239.32.21 is ###.###.220.23
[*] Running module against 216.239.34.21
[+] Source ip to 216.239.34.21 is ###.###.220.23
[*] Running module against 216.239.36.21
[+] Source ip to 216.239.36.21 is ###.###.220.23
[*] Auxiliary module execution completed
msf6 auxiliary(gather/external_ip) >
```